src/test/java/org/jenkinsci/plugins/build_token_root/BuildRootActionTest.java

```diff
@@ -112,6 +114,28 @@ private void assertCreated(Page page) throws Exception {
         j.waitUntilNoActivity();
     }
 
+    @Test public void responseStatus() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+        p.getBuildersList().add(new Shell("sleep 2"));
+        setAuthToken(p);
+        JenkinsRule.WebClient wc = j.createWebClient();
+        String buildUrl = "buildByToken/build?job=" + p.getFullName() + "&token=secret";
+        // Queue first build.
+        WebResponse resp1 = wc.goTo(buildUrl, "").getWebResponse();
+        // Queue second build, first is executing.
+        WebResponse resp2 = wc.goTo(buildUrl, "").getWebResponse();
+        // First is still executing, second is in the build queue, nothing to do.
+        WebResponse resp3 = wc.goTo(buildUrl, "").getWebResponse();
+        assertEquals(HttpURLConnection.HTTP_CREATED, resp1.getStatusCode());
+        assertTrue(resp1.getResponseHeaderValue("Location").contains("/queue/item/"));
+        assertEquals(HttpURLConnection.HTTP_CREATED, resp2.getStatusCode());
+        assertTrue(resp2.getResponseHeaderValue("Location").contains("/queue/item/"));
+        assertEquals(HttpURLConnection.HTTP_ACCEPTED, resp3.getStatusCode());
+        assertTrue(resp3.getResponseHeaderValue("Location").contains("/queue/item/"));
+        j.waitUntilNoActivity();
```

Comment:
With the above tip, you might still need this method, but basically just to wait after a web response until a build has been scheduled and you can get either a `Run` (already started) or `Queue.Item` (scheduled) for it. (Most tests would more directly call https://javadoc.jenkins.io/jenkins/model/ParameterizedJobMixIn.html#scheduleBuild2(int,hudson.model.Action...) but f course this plugin is about an unusual triggering method.) Possibly helpful APIs, off the top of my head:

* https://javadoc.jenkins.io/hudson/model/Queue.html#getItem(hudson.model.Queue.Task)
* https://javadoc.jenkins.io/hudson/model/Job.html#getBuildByNumber(int)
* https://javadoc.jenkins.io/hudson/model/queue/QueueTaskFuture.html#waitForStart()
* https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html#waitForCompletion-R-
* https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html#assertBuildStatusSuccess-java.util.concurrent.Future-